File: src/main/java/com/vaadin/demo/component/verticallayout/VerticalLayoutVerticalAlignment.java
package com.vaadin.demo.component.verticallayout;

import com.vaadin.demo.DemoExporter; // hidden-source-line
import com.vaadin.flow.component.html.Div;
import com.vaadin.flow.component.orderedlayout.FlexComponent;
import com.vaadin.flow.component.orderedlayout.VerticalLayout;
import com.vaadin.flow.router.Route;

@Route("vertical-layout-vertical-alignment")
public class VerticalLayoutVerticalAlignment extends Div {

    public VerticalLayoutVerticalAlignment() {
        // tag::layout[]
        VerticalLayout layout = new VerticalLayout();
        layout.setJustifyContentMode(FlexComponent.JustifyContentMode.CENTER);

        Div item1 = new Div("Item 1");
        item1.setClassName("example-item");

        Div item2 = new Div("Item 2");
        item2.setClassName("example-item");

        Div item3 = new Div("Item 3");
        item3.setClassName("example-item");

        layout.add(item1, item2, item3);
        // end::layout[]

        this.setClassName("basic-layouts-example");
        layout.setClassName("height-4xl");
        this.add(layout);
    }

    public static class Exporter extends // hidden-source-line
            DemoExporter<VerticalLayoutVerticalAlignment> { // hidden-source-line
    } // hidden-source-line
}

File: frontend/demo/component/vertical-layout/vertical-layout-vertical-alignment.ts
import 'Frontend/demo/init'; // hidden-source-line
import '@vaadin/vertical-layout';
import { html, LitElement } from 'lit';
import { customElement } from 'lit/decorators.js';
import { applyTheme } from 'Frontend/demo/theme';

@customElement('vertical-layout-vertical-alignment')
export class Example extends LitElement {
  protected override createRenderRoot() {
    const root = super.createRenderRoot();
    applyTheme(root);
    return root;
  }

  protected override render() {
    return html`
      <div class="basic-layouts-example">
        <!-- tag::snippet[] -->
        <vaadin-vertical-layout
          theme="spacing padding"
          class="height-4xl"
          style="justify-content: center"
        >
          <div class="example-item">Item 1</div>
          <div class="example-item">Item 2</div>
          <div class="example-item">Item 3</div>
        </vaadin-vertical-layout>
        <!-- end::snippet[] -->
      </div>
    `;
  }
}
